api/filrewardsd/pb/filrewards.proto

```diff
@@ -0,0 +1,126 @@
+syntax = "proto3";
+package api.filrewardsd.pb;
+option go_package = "github.com/textileio/textile/v2/api/filrewardsd/pb";
+
+import "google/protobuf/timestamp.proto";
+import "api/analyticsd/pb/analytics.proto";
+
+enum RewardType {
+  REWARD_TYPE_UNSPECIFIED = 0;
+  REWARD_TYPE_FIRST_KEY_ACCOUNT_CREATED = 1;
+  REWARD_TYPE_FIRST_KEY_USER_CREATED = 2; // does this fit "register first user?"
+  REWARD_TYPE_FIRST_ORG_CREATED = 3;
+  REWARD_TYPE_INITIAL_BILLING_SETUP = 4;
+  REWARD_TYPE_FIRST_BUCKET_CREATED = 5;
+  REWARD_TYPE_FIRST_BUCKET_ARCHIVE_CREATED = 6;
+  REWARD_TYPE_FIRST_MAILBOX_CREATED = 7;
+  REWARD_TYPE_FIRST_THREAD_DB_CREATED = 8;
+}
+
+message Reward {
+  string org_key = 1;
+  string dev_key = 2;
+  RewardType type = 3;
+  int64 factor = 4;
+  int64 base_nano_fil_reward = 5;
+  google.protobuf.Timestamp created_at = 6;
+}
+
+enum ClaimState {
+  CLAIM_STATE_UNSPECIFIED = 0;
+  CLAIM_STATE_PENDING = 1;
+  CLAIM_STATE_COMPLETE = 2;
+  CLAIM_STATE_FAILED = 3;
+}
+
+message Claim {
+  string id = 1;
+  string org_key = 2;
+  string claimed_by = 3;
+  int64 amount = 4;
+  ClaimState state = 5;
+  string txn_cid = 6;
+  string failure_message = 7;
+  google.protobuf.Timestamp created_at = 8;
+  google.protobuf.Timestamp updated_at = 9;
+}
+
+message ProcessAnalyticsEventRequest {
+  string org_key = 1;
+  string dev_key = 2;
+  api.analyticsd.pb.Event analytics_event = 3;
+}
+
+message ProcessAnalyticsEventResponse {
+  Reward reward = 1;
+}
+
+message ListRewardsRequest {
+  string org_key_filter = 1;
+  string dev_key_filter = 2;
+  RewardType reward_type_filter = 3;
+  bool ascending = 4;
+  int64 more_token = 5;
+  int64 limit = 6;
+}
+
+message ListRewardsResponse {
+  repeated Reward rewards = 1;
+  bool more = 2;
+  int64 more_token = 3;
+}
```

Comment:
cc @jsign this is what I settled on for communicating "more" and the needed token for the next requests. Idea is the token is, to the consumer, a meaningless integer they just need to pass in the next request. Turns out I am just using unix nanoseconds for the token, but the api doesn't need to be concerned with that.

Reply:
Any reason to not use Skip instead of this opaque token? I'm assuming that you're backed by MongoDB in this daemon.

Reply:
Just the potential for repeated records returned in the case that you sort descending (the default) and there are new records created in the time between page requests.

Reply:
Sounds good. The sorting criteria is always `CreatedAt` or som fixed and immutable date field?

Reply:
In this case, yes, always `CreatedAt`. I'm not claiming this is a universal strategy, just one that works in this case. Things change if more complex queries are supported.

Reply:
Yeah, sounds good... just exploring your case to double-check. I think for this scenario that looks fine, at least you can buy the consistency in the "pages" if some new things get created.

The "Skip-Limit" style seems more suited for some more variable sort scenario (despite you can get some drift in pages if things change while fetching next pages). "Skip" is also nice to skip pages and avoid forcing sequential exploration.

Taking this thread as a coffee talk to share ideas about paging patterns.

Reply:
Yea makes sense, I agree skip is the tool for general purpose paging, especially when allowing sorting by different keys.

Reply:
@asutula there are some existing APIs that do a similar thing: https://github.com/textileio/textile/blob/master/api/billingd/pb/billingd.proto#L73. The `next_offset` sounds a lot like your `more_token`. Seems like we should be aiming for consistency for things like this, even if these are internal APIs.

Reply:
Not suggesting a change here... just something to think about for future APIs

Reply:
Yea @sanderpick consistency is worth striving for, I agree, but also needs to be weighed by API correctness/accuracy. In this specific case (listing financial transactions basically), I went for the method that will never miss or skip any records (see comments about "drift" with pages above). If we want to make "skip" the consistent standard, I could remove the option for sorting descending and ascending sorting would be the only option (which could remove the drift problem).

Reply:
The sorting options sound useful. All good.